playground/miden-wasm/src/utils_debug.rs

```diff
@@ -140,57 +145,42 @@ impl DebugExecutor {
     // --------------------------------------------------------------------------------------------
 
     /// print general VM state information.
-    pub fn print_vm_state(&self) -> String {
-        format!("{}", self.vm_state)
-    }
+    fn vm_state_to_output(&self) -> DebugOutput {
+        let mut asmop = AsmOpInfo::new(String::new(), 0, 0);
+        if self.vm_state.asmop.is_some() {
+            asmop = self.vm_state.asmop.clone().unwrap();
+        }
 
-    /// print all stack items.
-    pub fn print_stack(&self) -> String {
-        let mut res =
-            self.vm_state
-                .stack
-                .iter()
-                .enumerate()
-                .fold(String::new(), |mut s, (i, f)| {
-                    s.push_str(&format!("[{i}] {f}\n"));
-                    s
-                });
-        res.pop(); // removes unnecessary line-break
-        res
+        let memory: Vec<(u64, [u64; 4])> =
+            self.vm_state.memory.iter().map(|x| (x.0, self.word_to_ints(&x.1))).collect();
+
+        let output = DebugOutput {
+            clk: self.vm_state.clk,
+            op: format!{"{:?}", self.vm_state.op},
```

Comment:
1. Let stick to the same naming as in the VM. The assembly code has instructions, the virtual machine executes operations.

So this should be `instruction`.

2. It seems `op` is a `Option`, so this formats like `None` or `Some(Inc)`. I think we change `DebugOutput.op` to be `Option<String>` and this line should be:

```rust
self.vm_state.op.map(|v| format!("{:?}" v))
```

Then the JS should handle the `None` case.

Reply:
Nice feedback, this is super helpful

So the VM has operations and Assembly Instructions (which is not quite consistently used when I look at `asmop`): which leads to the following table; 

```
clock: 3100
stack: [9,31,1,0,1,0,3,1,31,10,50,29,23,19,17,13,11,7,5,3,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
VM Operation: Some(MovUp3)
Assembly Instruction: gt
Instruction num of cycles: 18
Instruction cycle index: 13
Memory (Addr, Mem): [0]: [2, 0, 0, 0] 
                    [1]: [3, 0, 0, 0] 
```